mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/util/generation/GeneratedCodePatterns.java

```diff
@@ -117,15 +117,24 @@ public static void bindParameters(
     for (int i = 0; i < parameters.size(); i++) {
       VariableElement parameter = parameters.get(i);
       String parameterName = parameter.getSimpleName().toString();
-      String cqlName = (bindMarkerNames == null) ? parameterName : bindMarkerNames.get(i);
       PropertyType type = PropertyType.parse(parameter.asType(), context);
-      setValue(
-          cqlName,
-          type,
-          CodeBlock.of("$L", parameterName),
-          "boundStatementBuilder",
-          methodBuilder,
-          enclosingClass);
+      if (bindMarkerNames == null) {
+        setValue(
+            parameterName,
+            type,
+            CodeBlock.of("$L", parameterName),
+            "boundStatementBuilder",
+            methodBuilder,
+            enclosingClass);
+      } else {
+        setValue(
+            bindMarkerNames.get(i),
```

Comment:
why does that change? I think we were supporting bind marker names previously, aren't we?

Reply:
Nothing changes, it's just that now in some cases the bind marker names can be code blocks (when we retrieve them from the entity definition), and in others they're plain strings (when we use the Java parameter names as-is).

Reply:
For example, in this case:
```java
@Delete(entityClass = Product.class)
void deleteById(UUID productId);
```
`productId` could be called anything really, we don't care about the name because by looking at the entity definition, we know the bind marker will be called `id`. So we just say "bind the first parameter (whatever its name is), to `id`".
This corresponds to the branch `bindMarkerNames != null` in `bindParameters`, and now these names are code blocks.

In this case:
```java
@Select(customWhereClause = "WHERE description LIKE :searchString")
PagingIterable<Product> findByDescription(String searchString);
```
We don't know at compile time what the name of the bind marker will be, because it's in the user-provided string. So we require that it matches exactly: "bind the parameter called `searchString` to a placeholder called `searchString`".

Reply:
- Ok, isn't that first case of deleting a bit dangerous?
I mean that user can add a different name for that argument purposely (for example non-PK column) because he is new to Cassandra and doesn't know that deleting by non-pk is not possible. Then when issuing delete with that different argument he may get surprised that entity was not deleted - it may lead to weird problems (assuming that this non-PK key has the same type as inferred PK)

- We are doing type check, right? - so if the name of the argument is different/or the same as PK but the type is not matching PK what will happen?

Reply:
That's a bit far-fetched though, nowhere else in the mapper do we generate conditions on specific columns based on the name of a parameter.
Beyond that, I would say reading the docs and writing unit tests should prevent this kind of error.

If the types don't match, we generate an [error](https://github.com/datastax/java-driver/blob/java2078/mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/DaoDeleteMethodGenerator.java#L121-L134).